lib/vanagon/platform/windows.rb

```diff
@@ -53,9 +53,40 @@ def generate_packaging_artifacts(workdir, name, binding)
       # @param name [String] name of the project
       # @param binding [Binding] binding to use in evaluating the packaging templates
       def generate_msi_packaging_artifacts(workdir, name, binding)
-        FileUtils.mkdir_p(File.join(workdir, "wix"))
-        erb_file(File.join(VANAGON_ROOT, "resources/windows/wix/project.wxs.erb"), File.join(workdir, "wix",  "#{name}.wxs"), false, { :binding => binding })
-        erb_file(File.join(VANAGON_ROOT, "resources/windows/wix/project.filter.xslt.erb"), File.join(workdir, "wix", "#{name}.filter.xslt"), false, { :binding => binding })
+        # Recurse through the list of files in the resources/windows/wix directory and copy them over to the work directory.
+        copy_wix_artifacts_to_workdir("resources/windows/wix", workdir, name, binding)
+        copy_wix_artifacts_to_workdir(File.join(VANAGON_ROOT, "resources/windows/wix"), workdir, name, binding)
+      end
+
+      # Method to copy files over to the wix working directory from both the product
+      # specific directories (e.g. puppet-agent) and the generic (vanagon)
+      #
+      # @param srcdir [String] Originating directory
+      # @param workdir [String] working directory to stage the evaluated templates in
+      # @param name [String] name of the project
+      # @param binding [Binding] binding to use in evaluating the packaging templates
+      def copy_wix_artifacts_to_workdir(srcdir, workdir, name, binding)
+        Dir.chdir(srcdir) do
+          Dir.glob("**/*.*").each do |source_file|
+            src_basename = File.basename(source_file)
+            target_dir = File.dirname(File.join(workdir, "wix", File.dirname(source_file)))
+            FileUtils.mkdir_p(target_dir)
+
+            # ERB files are translated, all others are copied
+            if src_basename.end_with?(".erb")
+              target_file = File.join(target_dir, File.basename(src_basename, ".erb"))
+              copytransform = "erb_file(source_file, target_file, false, { :binding => binding })"
+            else
+              target_file = File.join(target_dir, src_basename)
+              copytransform = "FileUtils.cp(source_file, target_file, :verbose => true)"
+            end
+            if File.exists?(target_file)
+              puts "Target File #{target_file} already exists - Won't copy or transform #{source_file}"
+            else
+              eval copytransform
```

Comment:
When `instance_eval` is used in Vanagon,  it's to implement very concrete parts of the underlying patterns. Standard `eval` is a [Big Hammer](https://edwardsozlap.files.wordpress.com/2013/05/big-lap1-018.jpg) and it's commonly used to paper over flaws in the underlying assumptions that led to it being used. The O'Reilly book "Ruby Best Practices" actually has a [pretty good writeup](http://archive.oreilly.com/pub/a/ruby/excerpts/ruby-best-practices/worst-practices.html) about the dangers of `eval` (and its slightly less terrifying, slightly more justifiable cousin `instance_eval`).